omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/ManageAppsPageController.java

```diff
@@ -31,27 +31,33 @@
 
 @Controller
 public class ManageAppsPageController {
+	public List<String> appsThatCannotBeStopped = new ArrayList<String>();
+	public ManageAppsPageController(){
+		appsThatCannotBeStopped.add("coreapps.systemAdministrationApp");
+	}
 	
 	public void get(PageModel model, @SpringBean("appFrameworkService") AppFrameworkService service) {
 		addModelAttributes(model, service);
 	}
 	
-	public String post(PageModel model, @RequestParam("id") String id, @RequestParam("action") String action,
+	public String post(PageModel model, @RequestParam("id") String id, @RequestParam("action") String action, 
 	                   @SpringBean("appFrameworkService") AppFrameworkService service, HttpSession session, UiUtils ui) {
-		
 		String successMsgCode = "referenceapplication.app.manageApps." + action + ".success";
 		String failMessageCode = "referenceapplication.app.manageApps." + action + ".fail";
 		try {
 			if ("enable".equals(action)) {
 				service.enableApp(id);
-			} else if ("disable".equals(action)) {
+			} else if ("disable".equals(action) && !appsThatCannotBeStopped.contains(id)) {
 				service.disableApp(id);
 			} else if ("delete".equals(action)) {
 				service.purgeUserApp(service.getUserApp(id));
 			}
-			
-			InfoErrorMessageUtil.flashInfoMessage(session, ui.message(successMsgCode, id));
-			
+			if(!appsThatCannotBeStopped.contains(id)){
+				InfoErrorMessageUtil.flashInfoMessage(session, ui.message(successMsgCode, id));
+			}
+			else{
+				session.setAttribute(UiCommonsConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE, ui.message(failMessageCode, id));
```

Comment:
Why are we we allowing them to click on the lock icon, anyway? We should just have the lock not be clickable, instead of letting them do something illegal and showing an error message.